e2e/setup/serializers/image/diff-builder.ts

```diff
@@ -0,0 +1,43 @@
+import sharp from 'sharp';
+import type {SharpContext} from './snapshot-matcher';
+
+type DiffBuilderConfig = {
+  diffPath: string;
+  img1: SharpContext;
+  img2: SharpContext;
+  diffBuffer: Buffer;
+};
+
+export class DiffBuilder {
+
+  protected imgInfo: {width: number, height: number};
+
+  constructor(protected config: DiffBuilderConfig) {
+    this.imgInfo = this.config.img1.info;
+  }
+
+  public async build(): Promise<void> {
```

Comment:
Cosmetic (again that's internal implementation so no big concerns ):
The `build` function usually is immutable, while the current one have side effects (save to file)

Consider to rename class + function,
or execute final action outside class  
```ts